spec/unit/provider/oneview_san_manager_synergy_spec.rb

```diff
@@ -91,42 +89,41 @@
       end
 
       it 'should create/add the san manager' do
-        expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([])
-        expect(resourcetype).to receive(:find_by).with(anything, 'providerDisplayName' => resource['data']['providerDisplayName'])
-          .and_return([])
+        expect(resource_type).to receive(:find_by).and_return([])
+        expect_any_instance_of(resource_type).to receive(:retrieve!).and_return(false)
+        allow_any_instance_of(resource_type).to receive(:add).and_return(test)
         provider.exists?
-        allow_any_instance_of(resourcetype).to receive(:add).and_return(test)
         expect(provider.create).to be
       end
 
       it 'should update the san manager' do
-        expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([])
-        expect(resourcetype).to receive(:find_by).with(anything, 'providerDisplayName' => resource['data']['providerDisplayName'])
-          .and_return([test])
+        expect(resource_type).to receive(:find_by).and_return([])
```

Comment:
.and_return([test]) suits better here, once the resource already exists and it's being updated. The test will pass anyway.